### ISSUE TITLE
create_route53_record=false still try to create a record for ACM

### ISSUE THREAD

**Issue by bakayolo** [terraform-aws-modules/terraform-aws-atlantis#360]
## Description

Context: I am deploying Atlantis behind Cloudflare. I don't wanna use route53. And I understand that I'll have to set the records for validating certificates in Cloudflare by myself (I am doing this manually).
 
In my modules, I set 

```
route53_private_zone       = false
create_route53_aaaa_record = false
create_route53_record      = false
```

Note: `create_route53_record` is the only not default value.

The issue is happening in [this dependency module](https://github.com/terraform-aws-modules/terraform-aws-atlantis/blob/master/main.tf#L415-L425) and exactly in [this resource](https://github.com/terraform-aws-modules/terraform-aws-acm/blob/v3.2.0/main.tf#L34-L49).

I think we should find a way to provide `var.validate_certificate` in the module directly.
We could set `validate_certificate = var.create_route53_record || var.create_route53_aaaa_record` since I would assume that having both false we don't want to use route53 at all (which is my case) or we can create a new variable.
Note that I can also set `create_certificate = false` by passing my own certificate but it feels more like a workaround in that case.

Wdyt?

If your request is for a new feature, please use the `Feature request` template.

- [X] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version [Required]: `3.28.0`

- Terraform version: `v1.4.2`

- Provider version(s): `~> 4.0`

## Reproduction Code [Required]

Not adding any code since the issue is pretty obvious and described above.

## Expected behavior

I would expect a new certificate to be created, and have to validate it by myself (which is what I am already doing today fwiw).

## Actual behavior

Plan is erroring with

```
╷
│ Error: zone_id must not be empty, got 
│ 
│   with module.atlantis.module.atlantis.module.acm.aws_route53_record.validation[0],
│   on .terraform/modules/atlantis.atlantis.acm/main.tf line 37, in resource "aws_route53_record" "validation":
│   37:   zone_id = var.zone_id
│ 
╵
```

**Comment by bakayolo** [comment]
Actually, creating a certificate as part of the same deployment code using `aws_acm_certificate` resource and `certificate_arn` input in atlantis module does not work.

```
The "count" value depends on resource attributes that cannot be determined
│ until apply, so Terraform cannot predict how many instances will be
│ created. To work around this, use the -target argument to first apply only
│ the resources that the count depends on.
```

**Comment by antonbabenko** [comment]
This issue has been resolved in version 4.0.0 :tada:
